### DISCUSSION TITLE
Wrapping Database Queries (Entity Framework via DbContext) in Process Actor?

### DISCUSSION THREAD

**Discussion by rhemsuda** [louthy/echo-process#117] [Q&A] [answered]
Hey Paul, just found this library of yours and wondering if you would suggest using this as an immutable layer over my database? Currently I am just communicating with EntityFramework directly from functions wrapped in the Eff monad, but wondering if you see any downsides to that and if you'd suggest using this instead? I just watched your video on LanguageExt on YouTube and you mentioned this library was "a little bit broken" but this was 3 years ago so wondering what the status is here and if you can provide me any context into how this compares with Eff from LanguageExt 😄

**Comment by louthy** [comment, marked as answer]
I'm not currently actively supporting the library (although I haven't officially decided whether the deprecate it or not).  

I haven't migrated in to lang-ext v5, so for now I'd avoid it. If I ever get time I'd like to rewrite/refactor it, it evolved organically and is a little messy. It may well factor into my new startup, but I haven't decided yet.

However, the general idea of using actors (either in-memory or distributed) is that they're an extremely good pairing with FP.  

The actor inbox working as a fold over a stream of messages is cognitively easy to understand and the supervision hierarchy gives an elegant predictability to application structure, scale, parallelism, and failure handling.

In many ways actors are object-orientation done properly; with a functional world within each actor.

So, I would, for sure, advise the use of an actor library, just avoid my one for now.
